server/handlers/meshery_pattern_handler.go

```diff
@@ -889,19 +894,29 @@ func genericHTTPDesignFile(fileURL, sourceType string, reg *meshmodel.RegistryMa
 		return nil, ErrRemoteApplication(err)
 	}
 
-	k8sres := string(body)
+	res := string(body)
 
 	if sourceType == string(models.DockerCompose) {
-		k8sres, err = kompose.Convert(body)
+		res, err = kompose.Convert(body)
 		if err != nil {
 			return nil, ErrRemoteApplication(err)
 		}
 	}
 
-	pattern, err := pCore.NewPatternFileFromK8sManifest(k8sres, false, reg)
-	if err != nil {
-		return nil, err //This error is already a meshkit error
+	var pattern pCore.Pattern
+	if sourceType == string(models.DockerCompose) || sourceType == string(models.K8sManifest) {
```

Comment:
This should be removed and enhanced to use introspect

Reply:
At this point, have we already assessed what the source type is? If so, no introspection is needed.